src/infer_request.h

```diff
@@ -679,10 +680,32 @@ class InferenceRequest {
     secondary_stats_aggregator_ = secondary_stats_aggregator;
   }
 
-  void Cancel() { response_factory_->Cancel(); }
-  void ResetCancel() { response_factory_->ResetCancel(); }
+  Status Cancel()
+  {
+    if (response_factory_) {
```

Comment:
Question: if I understand this was also an issue in the previous iteration if Cancel() were called before SetResponseCallback() - is that correct?

Question: if a user calls inferasync without calling setresponsecallback - is that allowed?